rust/lance-index/src/vector/v3/subindex.rs

```diff
@@ -0,0 +1,62 @@
+// SPDX-License-Identifier: Apache-2.0
+// SPDX-FileCopyrightText: Copyright The Lance Authors
+
+use arrow_array::{RecordBatch, StructArray};
+use arrow_schema::SchemaRef;
+use lance_core::Result;
+use num_traits::Num;
+use roaring::RoaringBitmap;
+
+use super::storage::VectorStore;
+
+/// A sub index for IVF index
+pub trait IvfSubIndex<T: Num, Store: VectorStore>: Send + Sync + Sized {
+    type QueryParams: Default;
+
+    fn index_name(&self) -> &str;
+
+    /// Search the sub index for nearest neighbors.
+    /// # Arguments:
+    /// * `query` - The query vector
+    /// * `k` - The number of nearest neighbors to return
+    /// * `params` - The query parameters
+    /// * `pre_filter_bitmap` - The pre filter bitmap indicating which vectors to skip
+    fn search(
+        &self,
+        query: &[T],
+        k: usize,
+        params: Self::QueryParams,
+        storage: &Store,
+        pre_filter_bitmap: Option<RoaringBitmap>,
+    ) -> Result<RecordBatch>;
+
+    // check if the builder supports the metadata schema requested
```

Comment:
Not sure what does it mean?

Reply:
deleted, left over code I didn't clean up